Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -178,6 +184,7 @@ public func expandAttachedMacroWithoutCollapsing<Context: MacroExpansionContext>
   attributeNode: AttributeSyntax,
   declarationNode: DeclSyntax,
   parentDeclNode: DeclSyntax?,
+  extendedType: TypeSyntax?,
```

Comment:
I actually had the same thought while I was implementing this! But @rintaro pointed out to me that we might also need to make the `extendedType` an associated value of `PluginMessage.MacroRole` which would complicate the `Codable` conformance, so it seemed more straightforward to pass `extendedType` as an argument to `expandAttachedMacro`. I'm still interested in exploring this approach further separately though, because it seems plausible that we might want more role-specific arguments, and those conceptually make sense as enum associated values.